### PR TITLE
Clean up the data/ folder

### PR DESCRIPTION
**Pull request by hawkrives**
Whoof. Um. I moved things around?

> @hawkrives: How should I describe this PR?
> @drewvolz: A brighter future / Preparing for the cloud / API here we come

You'll want to read this introduction to YAML at some point: http://camel.readthedocs.io/en/latest/yamlref.html

---

The core of these changes is in a reorganization of the single-file structure of the data/ folder.

Before:

	images/
	about.md
	building-hours.json
	bus-times.json
	cage-menu.json
	calendar.json
	contact-info.json
	credits.json
	dictionary.json
	dietary-filters.js
	legal.json
	map-coordinates.json
	map.json
	new-building-hours.js
	news.json
	privacy.json
	transportation.json
	webcams.json

After: 

	building-hours/
	bus-times/
	contact-info/
	dictionary/
	map-coordinates/
	transportation/
	webcams/
	breaks.yaml
	chapel.yaml
	credits.yaml
	faqs.md
	legal.md
	map.yaml
	pause-menu.yaml
	privacy.md

I took the data files that were merely an array of objects, such as dictionary, building-hours, or bus-times, and converted them into folders with one-file-per-item, with each file converted to YAML.

The other data files I determined to be either plain text, or semi-structured stuff. I really really dislike editing JSON by hand, so the semi-structured stuff I converted to YAML files; the plain text ones I changed to Markdown.

(I added `js-yaml` and `marked` as devDependencies, since they're run at development time and not at runtime.)

When you run `npm run bundle-data`, which you should do after editing a data file, it automatically runs through the entire data/ folder and compiles them to docs/.

Individual files within a folder are compiled to a JSON array named after the folder – filenames are irrelevant, except to order the items in the generated array, as `{data: []}`.

Standalone files in data/ are compiled to one of two styles of output: either `{text: blah}` for a markdown input (e.g., legal.json), or `{data: blah}` for a yaml input (e.g., credits.json).

---

The main benefit, as I see it, to storing the files separately is that git is far better at tracking file changes in individual files than in one big blob.

I implemented a check in Travis to make sure that the data files have been regenerated in a PR, to make sure we don't accidentally ship old generated files.

---

The other change in this PR is that I moved the images from /data/images to just /images.

---

I feel that this PR sets us up nicely for putting more of our content on GitHub and fetching it dynamically in the app, like the Pause menu does.

---

TL;DR: go read the stuff and give me some feedback. I've got an FAQ view ready to go, too.

And finally some stats:

Changes in terms of lines changed, by folder:
```console
$ git diff --dirstat=changes,0 master
   0.0% views/
   0.0% views/dictionary/
   0.0% views/map/
   0.0% views/menus/
   0.0% views/menus/components/
   0.0% views/news/
   0.0% views/streaming/
   0.0% views/transportation/
   0.0% views/transportation/bus/__tests__/
   0.1% data/webcams/
   0.1% views/transportation/bus/
   0.2% data/images/
   0.3% data/transportation/
   0.3% views/settings/
   0.5% data/contact-info/
   0.5% images/
   0.6% views/building-hours/
   0.9% data/map-coordinates/
   1.4% scripts/
   1.5% docs/menus/
   1.8% data/bus-times/
   4.0% data/building-hours/
  13.5% data/dictionary/
  35.7% data/
  37.0% docs/
```

Changes in terms of files changed, by folder:
```console
$ git diff --dirstat=files,0 master
   0.2% data/images/
   0.2% docs/menus/
   0.2% views/
   0.2% views/dictionary/
   0.2% views/menus/
   0.2% views/menus/components/
   0.2% views/news/
   0.2% views/transportation/
   0.5% images/
   0.5% views/map/
   0.5% views/streaming/
   0.5% views/transportation/bus/__tests__/
   0.8% data/bus-times/
   0.8% views/settings/
   0.8% views/transportation/bus/
   1.4% data/contact-info/
   1.4% views/building-hours/
   1.7% data/webcams/
   1.7% scripts/
   2.0% data/transportation/
   4.6% docs/
   6.9% data/
  10.1% data/building-hours/
  17.1% data/map-coordinates/
  44.9% data/dictionary/
```

### CODE REVIEW

**Comment by hawkrives** [comment]
This PR depends on #474, because I moved the images folder in here and didn't update it in Xcode, and I don't think that I want to deal with conflicts between the two PRs.

Update: #474 has been merged. This is now free to merge. "You are now free to move about the cabin."

**Review by drewvolz — APPROVED** [review]

**Review by elijahverdoorn — APPROVED** [review]
I'm not reading through 420 files, but I get the point of this change and approve the ideas, so I'm trusting the implementation and approving.
